### PR TITLE
`auth api-info` command edits

### PR DESCRIPTION
**Pull request by cryptoquick**
**Summary of changes**
Changes introduced in this pull request:
- Make `auth api-info` behave more similar to how it does on Lotus, with regards to local defaults.
- Remove AuthApiInfo RPC API.

**Reference issue to close (if applicable)**

Closes 

**Other information and links**

Note: This is a PR into a PR.

An example of this using configuration to find the keystore file, and unlocking an unencrypted keystore. for the encrypted keystore command, I had restarted the daemon and used the encrypted keystore option in another window:

Screenshot from 2021-07-09 07-55-16

It also handles errors correctly, and overrides the encrypt_keystore configuration if the option is provided via the CLI:

Screenshot from 2021-07-09 07-55-16

### CODE REVIEW

File: forest/src/main.rs [hunk 1]
```diff
@@ -13,13 +13,16 @@ use structopt::StructOpt;
...
+    let CLI { opts, cmd } = CLI::from_args();
+
+    // Run forest as a daemon if no other subcommands are used. Otherwise, run the subcommand.
+    match opts.to_config() {
+        Ok(cfg) => match cmd {
+            Some(command) => subcommand::process(command, cfg).await,
+            None => daemon::start(cfg).await,
+        },
+        Err(e) => {
+            println!("Error parsing config. Error was: {}", e);
```
Comment (olibearo) [inline forest/src/main.rs:25]: Exit with an exit code `1` here. Could also print to stderr here

File: forest/src/subcommand.rs [hunk 2]
```diff
@@ -1,10 +1,63 @@
...
 /// Process CLI subcommand
-pub(super) async fn process(command: Subcommand) {
+pub(super) async fn process(command: Subcommand, config: Config) {
+    let token = match config.rpc_token.to_owned() {
+        Some(token) => token,
+        // If no token argument is passed or configured, attempt to load it from the local keystore
+        None => {
+            let keystore = if config.encrypt_keystore {
+                loop {
+                    println!("keystore passphrase: ");
```
Comment (olibearo) [inline forest/src/subcommand.rs:20]: Can we use a full sentence here and capitalize on new lines? `Enter the keystore passphrase: `

File: forest/src/subcommand.rs [hunk 3]
```diff
@@ -1,10 +1,63 @@
...
+
+                    let passphrase = read_password().expect("Error reading passphrase");
+
+                    let mut data_dir = PathBuf::from(&config.data_dir);
+                    data_dir.push(ENCRYPTED_KEYSTORE_NAME);
+
+                    if !data_dir.exists() {
+                        println!("keystore cannot be found from environment or arguments");
```
Comment (olibearo) [inline forest/src/subcommand.rs:28]: Capitalization

File: forest/src/main.rs (hunk 1 above)
Comment (cryptoquick) [reply forest/src/main.rs:25]: Good thinking!

**Review by connormullett — APPROVED** [review]
Looks good on my end, thanks for handling some of these fixes!

**Comment by olibearo** [comment]
@connormullett you should sync your branch with `main` so that this diff is more manageable

**Comment by connormullett** [comment]
> @connormullett you should sync your branch with `main` so that this diff is more manageable

I did prior to adding my review, but it didn't seem to help

**Comment by ec2** [comment]
image
I dont think I can review this PR if its in this state. I notice this is trying to merge into `connor/api-info` as opposed to `main`.

**Comment by ec2** [comment]
I think to fix this, since @connormullett has merged main into his branch, is to pull `connor/api-info` on top of this branch.

**Comment by cryptoquick** [comment]
> I think to fix this, since @connormullett has merged main into his branch, is to pull `connor/api-info` on top of this branch.

Done.

**Pull request merged**